src/circular_buffer.jl

```diff
@@ -49,17 +45,19 @@ Get the i-th element of CircularBuffer.
 * `cb[1]` to get the element at the front
 * `cb[end]` to get the element at the back
 """
-@inline Base.@propagate_inbounds function Base.getindex(cb::CircularBuffer, i::Int)
-    cb.buffer[_buffer_index_checked(cb, i)]
+Base.@propagate_inbounds function Base.getindex(cb::CircularBuffer, i::Int)
+    j = _buffer_index(cb, i)
+    @inbounds return cb.buffer[j]
 end
```

Comment:
If `i` is inbounds for `cb`, then `_buffer_index(cb, i)` is guaranteed inbounds for `cb.buffer`. We can separate these into two lines and annotate the second with `@inbounds`.

Also, `@inline` here is redundant because `@propagate_inbounds` implies `@inline`.